File: Year02/2nd semester/MPP/Lab/mpp-proiect-java-AlexandruPaul21/JavaProjectNetworking/ProjectNetworking/src/main/java/org/example/dto/DtoUtils.java
package org.example.dto;

import org.example.Agency;
import org.example.Booking;
import org.example.Flight;

import java.time.LocalDateTime;

public class DtoUtils {
    public static AgencyDto getDto(Agency agency) {
        String username = agency.getId();
        String name = agency.getName();
        String password = agency.getPassword();
        return new AgencyDto(username, name, password);
    }

    public static Agency fromDto(AgencyDto agency) {
        String username = agency.getUsername();
        String name = agency.getName();
        String password = agency.getPassword();
        return new Agency(username, name, password);
    }

    public static Flight fromDto(FlightDto flightDto) {
        Flight flight = new Flight(flightDto.getDestination(), LocalDateTime.parse(flightDto.getDeparture()), flightDto.getAirport(), Integer.parseInt(flightDto.getAvbSeats()));
        flight.setId(flightDto.getId());
        return flight;
    }

    public static FlightDto getDto(Flight flight) {
        return new FlightDto(flight);
    }

    public static BookingDto getDto(Booking booking) {
        return new BookingDto(booking.getId(), booking.getFlight(), booking.getClient(), booking.getPassengers());
    }

    public static Booking fromDto(BookingDto bookingDto) {
        Booking booking = new Booking(bookingDto.getFlight(), bookingDto.getClient(), bookingDto.getPassengers());
        booking.setId(bookingDto.getId());
        return booking;
    }

    public static Flight[] fromDto(FlightDto[] flightDtos) {
        Flight[] flights = new Flight[flightDtos.length];
        for (int i = 0; i < flightDtos.length; ++i) {
            flights[i] = fromDto(flightDtos[i]);
        }

        return flights;
    }

    public static FlightDto[] getDto(Flight[] flights) {
        FlightDto[] flightDtos = new FlightDto[flights.length];
        for (int i = 0; i < flights.length; ++i) {
            flightDtos[i] = getDto(flights[i]);
        }

        return flightDtos;
    }

}

File: Year02/2nd semester/MPP/Lab/mpp-proiect-csharp-AlexandruPaul21/NetworkingProject/networking/DtoUtils.cs
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace networking
{
    public class DtoUtils
    {
        public static AgencyDto getDto(Agency agency)
        {
            String username = agency.Id;
            String name = agency.Name;
            String password = agency.Password;
            return new AgencyDto(username, name, password);
        }

        public static Agency fromDto(AgencyDto agency)
        {
            String username = agency.username;
            String name = agency.name;
            String password = agency.password;
            var ag = new Agency(name, password);
            ag.Id = username;
            return ag;
        }

        public static Flight fromDto(FlightDto flightDto)
        {
            Flight flight = new Flight(flightDto.destination, DateTime.Parse(flightDto.departure), flightDto.airport, Int32.Parse(flightDto.avbSeats));
            flight.Id = flightDto.id;
            return flight;
        }

        public static FlightDto getDto(Flight flight)
        {
            return new FlightDto(flight);
        }

        public static BookingDto getDto(Booking booking)
        {
            return new BookingDto(booking.Id, booking.Flight, booking.Client, booking.Passengers);
        }

        public static Booking fromDto(BookingDto bookingDto)
        {
            Booking booking = new Booking(bookingDto.flight, bookingDto.client, bookingDto.passengers);
            booking.Id = bookingDto.id;
            return booking;
        }

        public static Flight[] fromDto(FlightDto[] flightDtos)
        {
            Flight[] flights = new Flight[flightDtos.Length];
            for (int i = 0; i < flightDtos.Length; ++i)
            {
                flights[i] = fromDto(flightDtos[i]);
            }

            return flights;
        }

        public static FlightDto[] getDto(Flight[] flights)
        {
            FlightDto[] flightDtos = new FlightDto[flights.Length];
            for (int i = 0; i < flights.Length; ++i)
            {
                flightDtos[i] = getDto(flights[i]);
            }

            return flightDtos;
        }
    }
}
